src/connectors/infinite-hits/connectInfiniteHits.ts

```diff
@@ -305,10 +304,6 @@ const connectInfiniteHits: InfiniteHitsConnector = function connectInfiniteHits(
       getWidgetState(uiState, { searchParameters }) {
         const page = searchParameters.page || 0;
 
-        if (!hasShowPrevious || !page) {
```

Comment:
Why did we need to remove the `!hasShowPrevious` condition?

Reply:
If we don't turn on `showPrevious`, `getWidgetState` always returns the uiState in this branch without setting any `page` key. And if there is no pagination widget in the widget tree, then uiState never get to have `page`. So "show more" didn't work. (example: https://codesandbox.io/s/instantsearchjs-dmuu2?file=/src/app.js)

Reply:
We seem to want to return `uiState` when the page comes from `showPrevious` and that it's lower than the current `uiState.page`.

The condition would look like this `!hasShowPrevious && page <= uiState.page || !page`.

Wdyt?

Reply:
Hey I tried your suggestion, but realized that in that function `uiState.page` is always undefined and that's why I removed that condition in the first place.